otter/test/convergence/test_selfheal.py

```diff
@@ -0,0 +1,328 @@
+"""
+Tests for :mod:`otter.convergence.selfheal`
+"""
+
+from effect.testing import SequenceDispatcher
+
+from kazoo.client import KazooClient
+from kazoo.exceptions import LockTimeout
+from kazoo.protocol.states import KazooState
+
+import mock
+
+from twisted.internet.defer import fail, succeed
+from twisted.internet.task import Clock
+from twisted.trial.unittest import SynchronousTestCase
+
+from otter.convergence import selfheal as sh
+from otter.log.intents import BoundFields
+from otter.models.intents import GetAllValidGroups, GetScalingGroupInfo
+from otter.models.interface import GroupState, ScalingGroupStatus
+from otter.test.utils import (
+    CheckFailure, const, intent_func, mock_log, nested_sequence, noop, patch,
+    perform_sequence, raise_)
+from otter.util.zk import GetChildren
+
+
+class SelfHealTests(SynchronousTestCase):
+    """
+    Tests for :obj:`SelfHeal`
+    """
+
+    def setUp(self):
+        self.lock = mock.Mock(specs=["acquire", "release"])
+        self.lock.release.return_value = succeed("released")
+        self.kzc = mock.Mock(specs=["Lock"])
```

Comment:
General tech-debt note: a functional test-mock for txkazoo would be very helpful in ensuring that tests are doing the Right Thing when it comes to the distributed lock semantics, instead of having each test spin up its own Mocks and hope that they do the right thing...